elyra/metadata/metadata_app.py

```diff
@@ -766,6 +766,120 @@ def start(self):
+        default_value=False,
+    )
+
+    # 'Import' flags
+    options: List[Option] = [directory_option, overwrite_flag]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.metadata_manager = MetadataManager(schemaspace=self.schemaspace)
+
+    def start(self):
+        super().start()  # process options
+
+        src_directory = self.directory_option.value
+
+        try:
+            json_files = [f for f in os.listdir(src_directory) if f.endswith(".json")]
+        except OSError as e:
+            print(f"Unable to reach the '{src_directory}' directory: {e.strerror}: '{e.filename}'")
+            self.exit(1)
+
+        if len(json_files) == 0:
+            print(f"No instances for import found in the '{src_directory}' directory")
+            return
+
+        metadata_file = None
+        non_imported_files = []
+
+        for file in json_files:
+            filepath = os.path.join(src_directory, file)
+            try:
+                with open(filepath) as f:
+                    metadata_file = json.loads(f.read())
+                f.close()
+            except OSError as e:
+                non_imported_files.append([file, e.strerror])
+                continue
+
+            name = os.path.splitext(file)[0]
+            try:
+                schema_name = metadata_file["schema_name"]
+                display_name = metadata_file["display_name"]
+                metadata = metadata_file["metadata"]
+            except KeyError as e:
+                non_imported_files.append([file, f"Could not find '{e.args[0]}' key in the import file '{filepath}'"])
+                continue
+
+            try:
+                if self.overwrite_flag.value:  # if overwrite flag is true
+                    try:  # try updating the existing instance
+                        updated_instance = self.metadata_manager.get(name)
+                        updated_instance.schema_name = schema_name
+                        if display_name:
+                            updated_instance.display_name = display_name
+                        if name:
+                            updated_instance.name = name
+                        updated_instance.metadata.update(metadata)
+                        self.metadata_manager.update(name, updated_instance)
+                    except Exception:  # no existing instance - create new
+                        instance = Metadata(
+                            schema_name=schema_name, name=name, display_name=display_name, metadata=metadata
+                        )
+                        self.metadata_manager.create(name, instance)
+                else:
+                    instance = Metadata(
+                        schema_name=schema_name, name=name, display_name=display_name, metadata=metadata
+                    )
+                    self.metadata_manager.create(name, instance)
+            except Exception as e:
+                non_imported_files.append([file, str(e)])
```

Comment:
I wonder if we should make a suggestion when `isinstance(e, MetadataExistsError)` is true that `--overwrite` can be used to update.  This might be helpful when the `--overwrite` flag is not used and there are existing valid instances, like the following output...
```
$ elyra-metadata import runtimes --directory /tmp/exports/runtimes
Imported 0 instances
5 instances could not be imported

The following files could not be imported: 
File                Reason                                                                                                                           
----                ------                                                                                                                           
af_teacloth1.json   Validation failed for instance 'af_teacloth1' using the airflow schema with error: 'teacloth1.fyre.ibm.com+32048' is not a 'uri'.
kf_cloning1.json    An instance named 'kf_cloning1' already exists in the runtimes schemaspace.                                                      
kf_devises1.json    An instance named 'kf_devises1' already exists in the runtimes schemaspace.                                                      
kf_eg3.json         An instance named 'kf_eg3' already exists in the runtimes schemaspace.                                                           
kfp-local.json      Validation failed for instance 'kfp-local' using the kfp schema with error: 'xyz--localhost+31380/pipeline' is not a 'uri'.   
```
If users weren't aware of `--overwrite` this might be a little worrisome, so perhaps appending instructions that would avoid that particular error might be helpful...
```
$ elyra-metadata import runtimes --directory /tmp/exports/runtimes
Imported 0 instances
5 instances could not be imported

The following files could not be imported: 
File                Reason                                                                                                                           
----                ------                                                                                                                           
af_teacloth1.json   Validation failed for instance 'af_teacloth1' using the airflow schema with error: 'teacloth1.fyre.ibm.com+32048' is not a 'uri'.
kf_cloning1.json    An instance named 'kf_cloning1' already exists in the runtimes schemaspace.  (use --overwrite to update)                                                      
kf_devises1.json    An instance named 'kf_devises1' already exists in the runtimes schemaspace.  (use --overwrite to update)                                                  
kf_eg3.json         An instance named 'kf_eg3' already exists in the runtimes schemaspace.  (use --overwrite to update)                                                           
kfp-local.json      Validation failed for instance 'kfp-local' using the kfp schema with error: 'xyz--localhost+31380/pipeline' is not a 'uri'.   
```

It looks a bit odd to have a parenthesized comment after the period, so it might look better to make the suggestion a sentence as well..
```
af_teacloth1.json   Validation failed for instance 'af_teacloth1' using the airflow schema with error: 'teacloth1.fyre.ibm.com+32048' is not a 'uri'.
kf_cloning1.json    An instance named 'kf_cloning1' already exists in the runtimes schemaspace.  Use --overwrite to update.                                                     
kf_devises1.json    An instance named 'kf_devises1' already exists in the runtimes schemaspace.  Use --overwrite to update.    
```

Reply:
I agree. That additional comment would be very useful to display.